### PR TITLE
Scripting brush ups including licensing

### PR DESCRIPTION
**Pull request by milanlakhani**
Added L1 and L2 requirement lists in yml.
Applied discussed changes to yml component files.

### CODE REVIEW

File: applying-stpa/gen_requirements.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import yaml
```
Comment (kestewart) [inline applying-stpa/gen_requirements.py]: Please add a license in SPDX format to the script.

File: applying-stpa/l2_reqs.txt [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+ ('REQ-102', 'OpenAPS must set up the algorithm that the human has entered in the setup script'),
+ ('REQ-103', 'OpenAPS should test and track an algorithms performance with a wide range of different config settings'),
+ ('REQ-104', 'OpenAPS must have a forum where people can inform developers of algorithm malfunctions'),
+ ('REQ-105', 'Human must make sure that algorithm has been tested properly before using it'),
+ ('REQ-106', 'OpenAPS must specify in the documentation how to make sure an algorithm is safe'),
+ ('REQ-107', 'OpenAPS should provide a service to test the safety of algorithms'),
+ ('REQ-108', 'OpenAPS should have a facility to allow the human to test an algorithm, while running a different algorithm, and compare dosing recommendations'),
+ ('REQ-109', 'When new settings have been set, OpenAPS must update based on these settings before the next loop'),
+ ('REQ-110', 'OpenAPS should have a fixed spot to check for and implement updates in every (~5 minute) OpenAPS loop'),
+ ('REQ-111', '(1) process topic: Data formats (APIs ?) that OpenAPS works with must be specified'),
```
Comment (kestewart) [inline applying-stpa/l2_reqs.txt]: Not sure about the syntax "(1)" process topic"  - should this be included in the requirements set?

**Review by kestewart — CHANGES_REQUESTED** [review]
Can you separate this into 2 pull requests please.   
One for the scripts,  and one for the outcome of running the scripts.

Also,   all python scripts should have licensing information in the top of them

File: applying-stpa/gen_requirements.py (hunk 1 above)
Comment (kestewart) [reply applying-stpa/gen_requirements.py]: Put in a python comment something like...
''' SPDX-License-Identifier: GPL-2.0 '''

File: applying-stpa/l2_reqs.txt [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+ ('REQ-28', 'OpenAPS developers shall specify the versions of RaspiOS which are support on each version of OpenAPS'),
+ ('REQ-29', 'OpenAPS app shall do a compatibility check on whether it is compatible with the current RaspiOS version'),
+ ('REQ-30', 'On boot, RaspiOS shall automatically start OpenAPS'),
+ ('REQ-31', 'The OpenAPS loop shall provide an "I\'m alive!" signal whenever it is running so the user knows it is active'),
+ ('REQ-32', 'OpenAPS itself shall check to see if there is a newer version of OpenAPS installed'),
+ ('REQ-33', 'OpenAPS shall check for multiple instances of it running'),
+ ('REQ-34', 'User shall have fresh install of Raspi OS on machine that meets minimum requirements with no other applications running except for OpenAPS'),
+ ('REQ-35', 'If RasPi OS forwards the data in the incorrect format, OpenAPS shall check to ensure that the data is in the correct format.  If RasPi OS does not forward the data, OpenAPS shall implement a timeout period to ensure it receives data in a timely manner.  If either check fails, OpenAPS shall alarm the user.'),
+ ('REQ-36', "(I need help with this one!  It's not under OpenAPS's control)"),
```
Comment (kestewart) [inline applying-stpa/l2_reqs.txt]: Let's do a pass after separate commitment.

File: applying-stpa/requirements.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+pandas
```
Comment (kestewart) [inline applying-stpa/requirements.txt]: Add comment to explain why done.

**Comment by milanlakhani** [comment]
> Can you separate this into 2 pull requests please. One for the scripts, and one for the outcome of running the scripts.
> 
> Also, all python scripts should have licensing information in the top of them

This is both done

**Comment by kestewart** [comment]
DIscussed on call today, and Milan will make an additional split.

**Comment by milanlakhani** [comment]
> DIscussed on call today, and Milan will make an additional split.

This should now be done

**Review by kestewart — APPROVED** [review]
Thanks Milan.  Looks good!

**Comment by kestewart** [comment]
Hmm... can you add your Signed-off-by:  into this thread, if not in the commit.    Then I'll merge.

**Review by Shefali199 — APPROVED** [review]

**Comment by milanlakhani** [comment]
Hi Kate, sorry for the delay, I've added sign offs to all my commits! I'll keep an eye on the CI. Also I am no longer to access the OpenAPS STPA spreadsheet, are you an owner who can change this? Otherwise it might be to do with the fact that originally this was linked to my Codethink address, I can e-mail CT about it.

**Pull request merged**